pkg/domain/infra/abi/play.go

```diff
@@ -1261,13 +1262,35 @@ func splitMultiDocYAML(yamlContent []byte) ([][]byte, error) {
 			return nil, fmt.Errorf("multi doc yaml could not be split: %w", err)
 		}
 
-		if o != nil {
-			// back to bytes
-			document, err := yamlv3.Marshal(o)
-			if err != nil {
-				return nil, fmt.Errorf("individual doc yaml could not be marshalled: %w", err)
+		if o == nil {
+			continue
+		}
+
+		// back to bytes
+		document, err := yamlv3.Marshal(o)
+		if err != nil {
+			return nil, fmt.Errorf("individual doc yaml could not be marshalled: %w", err)
+		}
+
+		kind, err := getKubeKind(document)
+		if err != nil {
+			return nil, fmt.Errorf("couldn't get object kind: %w", err)
+		}
+
+		if kind == "List" {
```

Comment:
Can you add a comment why lists need special casing?

Reply:
Done